### PR TITLE
feat: check sa before start a job

### PR DESCRIPTION
**Pull request by jiuker**
feat: check sa before start a job

### CODE REVIEW

File: pkg/controller/job-controller.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 	"github.com/minio/minio-go/v7/pkg/set"
 	"github.com/minio/operator/pkg/apis/job.min.io/v1alpha1"
 	miniov2 "github.com/minio/operator/pkg/apis/minio.min.io/v2"
+	v1alpha12 "github.com/minio/operator/pkg/apis/sts.min.io/v1alpha1"
```
Comment (shtripat) [inline pkg/controller/job-controller.go:14]: Why `v1alpha12` and not just `v1alpha1` ;)

File: pkg/controller/job-controller.go [hunk 2]
```diff
@@ -246,7 +247,24 @@ func (c *JobController) SyncHandler(key string) (Result, error) {
...
+	// check sa
+	pbs := &v1alpha12.PolicyBindingList{}
+	err = c.k8sClient.List(ctx, pbs, client.InNamespace(namespace))
+	if err != nil {
+		return WrapResult(Result{}, err)
+	}
+	if len(pbs.Items) == 0 {
+		return WrapResult(Result{}, fmt.Errorf("no policybinding found"))
+	}
+	haveSa := false
```
Comment (shtripat) [inline pkg/controller/job-controller.go:259]:
```suggestion
	saFound := false
```

File: pkg/controller/job-controller.go (hunk 1 above)
Comment (jiuker) [reply pkg/controller/job-controller.go:14]: already used for `github.com/minio/operator/pkg/apis/job.min.io/v1alpha1` @shtripat

Comment (shtripat) [reply pkg/controller/job-controller.go:14]: ah ok.

**Review by cniackz — APPROVED** [review]
Yeah, that make sense

**Pull request merged**